### PR TITLE
Notify users via email anyime they book a new recurring shift

### PR DESCRIPTION
**Pull request by apackin**
#### What's this PR do?
Emails any users that booked a shift after they have received their second_shift_notification (or are too old to)
#### How should this be manually tested?
 1) Change the runJobs file to only run this
 2) Book a new shift on whenIwork with an account that has already received second_shift_notification or more than the designated age
3) node runJobs 
4) make sure you get an email about your newly booked shift
#### What are the relevant tickets?
https://admin.crisistextline.org/jira/secure/RapidBoard.jspa?rapidView=5&view=detail&selectedIssue=INT-164

### CODE REVIEW

File: jobs/scheduling/helpers/notifyUserBookedShift.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+var CronJob = require('cron').CronJob;
```
Comment (jrauschenberg) [inline jobs/scheduling/helpers/notifyUserBookedShift.js]: You require this but don't use it.

File: jobs/scheduling/helpers/notifyUserBookedShift.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+var wIWUserAPI = CONFIG.WhenIWork;
```
Comment (jrauschenberg) [inline jobs/scheduling/helpers/notifyUserBookedShift.js]: Same as with line 1--required but not used.

File: jobs/scheduling/helpers/notifyUserBookedShift.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  });
+}
+
+function filterUsersToObject (users){
+  var usersObj = {};
+  users.forEach(function(user){
+    var created = moment(new Date(user.created_at));
+    // Only includes users who have already received the two_shift_notification note
+    // or are at least 'days_of_open_shift_display' old and will never get that user note
+    if (/two_shift_notification/.test(user.notes) || moment().diff(created, 'days') >= CONFIG.time_interval.days_of_open_shift_display) usersObj[user.id] = user;
```
Comment (jrauschenberg) [inline jobs/scheduling/helpers/notifyUserBookedShift.js]: I changed this in the config file per your suggestion, FYI, but that change hasn't been merged yet.

**Comment by jrauschenberg** [comment]
Made a few comments. Other than that, I can't get this to run on my machine and e-mail me either using your instructions or just running the file with node. I think because it has to get called from RecurShifts?

**Comment by apackin** [comment]
Resolved John's issues @tongxiang .

**Comment by tongxiang** [comment]
Good, clear instructions on how to test.

**Comment by tongxiang** [comment]
👍  good modularization in `notifyUserBookedShift`. Feel free to merge!

**Pull request merged**